pmd-apex/src/main/java/net/sourceforge/pmd/lang/apex/rule/bestpractices/UnusedLocalVariableRule.java

```diff
@@ -34,7 +34,7 @@ public class UnusedLocalVariableRule extends AbstractApexRule {
             "Database.countQuery".toLowerCase(Locale.ROOT)
     ));
 
-    private static final Pattern BINDING_VARIABLE = Pattern.compile("(?i):([a-z0-9]+)");
+    private static final Pattern BINDING_VARIABLE = Pattern.compile("(?i):(?:\s|)([_a-z0-9]+)");
```

Comment:
There could be more than one space
```suggestion
    private static final Pattern BINDING_VARIABLE = Pattern.compile("(?i):\s*+([_a-z0-9]+)");
```

Reply:
Also we should add a test case testing this in https://github.com/pmd/pmd/blob/master/pmd-apex/src/test/resources/net/sourceforge/pmd/lang/apex/rule/bestpractices/xml/UnusedLocalVariable.xml

Reply:
The spaces are all optional. I don't understand the syntax enough to know if that is accounted for in the patterns, even before `=` and after `:`.

Reply:
It's not a typo, it's a possessive quantifier. It will not backtrack so the regex will be faster.

Reply:
And here I was thinking I knew my regular expressions… thanks for that!